### PR TITLE
Use repos.json to access GOV.UK repositories

### PR DESCRIPTION
**Pull request by kevindew**
This uses a new endpoint, repos.json [1], to access the GOV.UK
repositories. The apps.json endpoint only returns running applications,
which means that Gems and similar non-running apps aren't including.

Switching to repos.json will include those projects

~~This is open as a draft whilst https://github.com/alphagov/govuk-developer-docs/pull/3481 is open.~~ I also found that I had problems running this due to secondary rate-limiting so it's been hard to test, not sure if there are any tips?

[1]: https://github.com/alphagov/govuk-developer-docs/pull/3481

### CODE REVIEW

**Review by ChrisBAshton — COMMENTED** [review]
LGTM. Afraid I don't have any tips RE the secondary rate limiting - is that an issue on `master` too or only on this branch? If the latter, we should probably look at fixing that before merge 😬 If the former, then that shouldn't hold up this PR.

**Comment by kevindew** [comment]
> LGTM. Afraid I don't have any tips RE the secondary rate limiting - is that an issue on `master` too or only on this branch? If the latter, we should probably look at fixing that before merge 😬 If the former, then that shouldn't hold up this PR.

Yup, I get the same on the master branch. Steps to replicate are run `bundle exec rackup` and then visit localhost:9292

**Comment by ChrisBAshton** [comment]
Hmmm I've tried exposing a `GITHUB_TOKEN` environment variable on `master` but unfortunately it hasn't helped.

This issue was supposedly fixed in #111 but that either didn't work, or a change has happened on GitHub's side.

**Comment by kevindew** [comment]
> Hmmm locally I've tried exposing a `GITHUB_TOKEN` environment variable on `master` but unfortunately it hasn't helped.
> 
> This issue was supposedly fixed in #111 but that either didn't work, or a change has happened on GitHub's side.

Sure, I don't think a cache would help the situation you face when starting the app locally - as you have a cold cache that needs to populate.

I did wonder if there was a quick half fix of seeing if you could reduce the repos that are pulled in https://github.com/alphagov/govuk-dependencies/blob/0f35d6003f0346d0aa7c14ad1e94f7a74b1587e8/lib/gateways/pull_request.rb#L16-L40 to just GOV.UK ones, as I understand it'll paginate through all of GDS' repos - however my approach of adding `topic:govuk` didn't seem to work. Though, as I say, that'd just be a half fix. I wonder if a different approach could be just filtering to repos we care about as I think it needs to be a matching repo to actually be shown.

I imagine an actual fix is more fundamental. A crude one could be that it builds data in the background with waiting between requests and doesn't do it on demand. I imagine the one that'd suit GitHub best is if this had a database and listened for GitHub events that stored the PRs rather than acting on demand - though that'd be a major rethink.

**Review by MuriloDalRi — APPROVED** [review]

**Comment by MuriloDalRi** [comment]
I've merged another PR to fix the rate-limiting issues: https://github.com/alphagov/govuk-dependencies/pull/120

This should work now locally but you won't be able to deploy to Heroku until those issues get sorted.

**Comment by MuriloDalRi** [comment]
I'll deploy this manually

**Pull request merged**
